src/app/reporting/Engine.cpp

```diff
@@ -127,6 +127,26 @@ std::optional<CHIP_ERROR> ValidateReadAttributeACL(DataModel::Provider * dataMod
     return err == CHIP_ERROR_ACCESS_DENIED ? CHIP_IM_GLOBAL_STATUS(UnsupportedAccess) : CHIP_IM_GLOBAL_STATUS(AccessRestricted);
 }
 
+/// Checks that the given attribute path corresponds to a readable attribute. If not, it
+/// will return the corresponding failure status.
+std::optional<Status> ValidateAttributeIsReadable(DataModel::Provider * dataModel, const ConcreteReadAttributePath & path)
+{
+    DataModel::AttributeFinder finder(dataModel);
+
+    std::optional<DataModel::AttributeEntry> entry = finder.Find(path);
+    if (!entry.has_value())
+    {
+        return DataModel::ValidateClusterPath(dataModel, path, Status::UnsupportedAttribute);
+    }
+
+    if (!entry->GetReadPrivilege().has_value())
+    {
+        return Status::UnsupportedAttribute;
```

Comment:
I would think this should be:

```suggestion
        return Status::UnsupportedRead;
```

but am I missing something?